src/pages/patientView/clinicalInformation/lib/clinicalAttributesUtil.js

```diff
@@ -207,21 +207,132 @@ function getSpanElementsFromCleanData(
+            OS_STATUS: 7,
+            OS_MONTHS: 8,
+            DFS_STATUS: 9,
+            DFS_MONTHS: 10,
+        };
+
+        return map[key] != undefined ? map[key] : Number.MAX_VALUE;
+    };
+
+    // Ascending order; e.g. GENDER precedes OS_MONTHS
+    const compare = (a, b) => {
+        return getOrder(a) - getOrder(b);
+    };
+
+    // Styles before and after value
+    function stringBuilder(value, key) {
+        var builder = [];
+
+        const before = {
+            /* attributes with opening parenthesis */
+            OS_MONTHS: '\xa0(',
+            DFS_MONTHS: '\xa0(',
+            CANCER_TYPE_DETAILED: '\xa0(',
+            PATIENT_DISPLAY_NAME: '\xa0(',
+            DERIVED_SAMPLE_LOCATION: '\xa0(',
+            SAMPLE_DISPLAY_NAME: '\xa0(',
+
+            /* text before an attribute */
+            SERUM_PSA: ', Serum PSA: ',
+            ERG_FUSION_ACGH: ', ERG-fusion aCGH: ',
+            TMPRSS2_ERG_FUSION_STATUS: ', TMPRSS2-ERG Fusion: ',
+            GLEASON: ', Gleason: ',
+        };
+
+        const after = {
+            /* attributes with a closing parenthesis */
+            CANCER_TYPE_DETAILED: ')',
+            PATIENT_DISPLAY_NAME: ')',
+            DERIVED_SAMPLE_LOCATION: ')',
+            SAMPLE_DISPLAY_NAME: ')',
+
+            /* text after an attribute */
+            OS_MONTHS: ' months)',
+            DFS_MONTHS: ' months)',
+            AGE: ' years old',
+        };
+
+        /* Show comma before clinical attributes */
+        // Prefix is never used, but its assignment allows the side effect of push to happen
+        const prefix =
+            before[key] == '\xa0(' ? builder.push('') : builder.push(', ');
+        builder.push(before[key]);
+        builder.push(value);
+        builder.push(after[key]);
+        return builder.join('');
+    }
+
+    function colorGetter(value, key) {
+        const colors = {
+            sample_color_primary: 'black',
+            sample_color_recurrence: 'orange',
+            sample_color_metastasis: 'red',
+            sample_color_cfdna: 'blue',
+            sample_color_xenograft: 'pink',
+        };
+
+        /* attributes with special colors */
+        const keys = {
+            OS_STATUS: {
+                DECEASED: '#f00',
+                DEAD: '#f00',
+                LIVING: 'rgb(0, 128, 0)',
+                ALIVE: 'rgb(0, 128, 0)',
+            },
+            DFS_STATUS: {
+                _: '#f00',
+                DiseaseFree: 'rgb(0, 128, 0)',
+                Yes: 'rgb(0, 128, 0)',
+            },
+            DERIVED_NORMALIZED_CASE_TYPE: {
+                Primary: colors.sample_color_primary,
+                Progressed: colors.sample_color_recurrence,
+                Recurrence: colors.sample_color_recurrence,
+                Metastasis: colors.sample_color_metastasis,
+            },
+        };
+
+        let color = 'black';
+        if (keys[key] != undefined) {
+            if (keys[key]['_'] != undefined) {
```

Comment:
Magic constants like `'_'` are a bit worrisome. What does `'_'` represent? The code might be more readable if you create a const near the top of the file `const NAME_THAT_EXPLAINS_VAR = '_';` and then use that here.